### PR TITLE
ASM-2835: Packaging changes to support graphite web server

### PR DESCRIPTION
**Pull request by ripienaar**
These changes are needed to add dependencies for graphite and to
configure it to work out of the box

DO NOT MERGE YET.  DEPENDS ON asm-deployer CHANGES NOT AVAILABLE YET

### CODE REVIEW

File: scripts/rpm/postInstall.sh [hunk 1]
```diff
@@ -6,3 +7,28 @@
 
 /sbin/restorecon -v /usr/lib64/nagios/plugins/*
 
+# configure graphite
```
Comment (ripienaar) [inline scripts/rpm/postInstall.sh]: This isnt particularly awesome, would be better in a puppet module - but we dont build the appliance that way so at least this way we can control stuff and this shouldnt break on updates

**Comment by ripienaar** [comment]
Relates to https://github.com/dell-asm/asm-deployer/pull/690, merge after that

File: build.gradle [hunk 2]
```diff
@@ -2,6 +2,9 @@ import groovy.text.GStringTemplateEngine
 import org.freecompany.redline.payload.*
 import org.freecompany.redline.header.*
 
+// https://github.com/nebula-plugins/gradle-ospackage-plugin/issues/59
```
Comment (gavin-scott) [inline build.gradle]: Interesting, this is good to know...

File: build.gradle [hunk 3]
```diff
@@ -64,6 +67,12 @@ task rpm() << {
     rpmBuilder.addDependencyMore('ipmitool', '1.8.11')
     rpmBuilder.addDependencyMore('pywbem', '0.7.0')
 
+    // Graphite dependencies
+    rpmBuilder.addDependencyMore('graphite-web', '0.9.12')
+    rpmBuilder.addDependencyMore('graphite-web-selinux', '0.9.12')
+    rpmBuilder.addDependencyMore('python-whisper', '0.9.12')
+    rpmBuilder.addDependencyMore('python-carbon', '0.9.12')
```
Comment (gavin-scott) [inline build.gradle]: Do we have all of these in our repos?

File: scripts/rpm/postInstall.sh [hunk 4]
```diff
@@ -6,3 +7,28 @@
...
+grep -q ^SECRET_KEY /etc/graphite-web/local_settings.py
+
+if [ $? -eq 1 ]
+then
+  echo "SECRET_KEY = '$(openssl rand 32 -hex)'" >> /etc/graphite-web/local_settings.py
```
Comment (gavin-scott) [inline scripts/rpm/postInstall.sh]: Yay, I appreciate having different keys on different appliance instances!

File: build.gradle (hunk 3 above)
Comment (ripienaar) [inline build.gradle]: Yes, I tested this installs cleanly when running the yum command by hand, tested various install/update scenarios etc

File: scripts/rpm/postInstall.sh [hunk 5]
```diff
@@ -6,3 +7,28 @@
...
+  echo "TIME_ZONE = 'UTC'" >> /etc/graphite-web/local_settings.py
```
Comment (gavin-scott) [inline scripts/rpm/postInstall.sh]: Any problems with always setting this to UTC? Will it be the times be user-visible?

Comment (ripienaar) [inline scripts/rpm/postInstall.sh]: I dont know whats best to be honest, this needs to be set to something and probably be in sync with the appliance.  But once the data files are created its best not to mess around with this stuff.

So I suspect we either have to massage the data as we retrieve it from graphite into local time in the REST end point or the javascript side should just deal with it but storing in UTC and not allowing it to be changed seems sanest

Comment (gavin-scott) [inline scripts/rpm/postInstall.sh]: Yeah, if changing the time zone would cause problems leaving it in UTC is probably the best we can do.

**Pull request merged**